// angular2-notifications/src/simple-notifications/components/notification.component.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as i0 from '@angular/core';
import * as i1 from '@angular/common';
import * as i2 from 'angular2-notifications/src/simple-notifications/pipes/max.pipe';
import * as i3 from 'angular2-notifications/src/simple-notifications/components/notification.component';
import * as i4 from 'angular2-notifications/src/simple-notifications/services/notifications.service';
import * as i5 from '@angular/platform-browser';
const styles_NotificationComponent:any[] = ['\n        .simple-notification {\n            width: 100%;\n            padding: 10px 20px;\n            box-sizing: border-box;\n            position: relative;\n            float: left;\n            margin-bottom: 10px;\n            color: #fff;\n            cursor: pointer;\n            transition: all 0.5s;\n        }\n\n        .simple-notification .sn-title {\n            margin: 0;\n            padding: 0 50px 0 0;\n            line-height: 30px;\n            font-size: 20px;\n        }\n\n        .simple-notification .sn-content {\n            margin: 0;\n            font-size: 16px;\n            padding: 0 50px 0 0;\n            line-height: 20px;\n        }\n\n        .simple-notification .icon {\n            position: absolute;\n            box-sizing: border-box;\n            top: 0;\n            right: 0;\n            width: 70px;\n            height: 70px;\n            padding: 10px;\n        }\n\n        .simple-notification .icon svg {\n            fill: #fff;\n            width: 100%;\n            height: 100%;\n        }\n\n        .simple-notification .icon svg g {\n            fill: #fff;\n        }\n\n        .simple-notification.rtl-mode {\n            direction: rtl;\n        }\n\n        .simple-notification.rtl-mode .sn-content {\n            padding: 0 0 0 50px;\n        }\n\n        .simple-notification.rtl-mode svg {\n            left: 0;\n            right: auto;\n        }\n\n        .simple-notification.error { background: #F44336; }\n        .simple-notification.success { background: #8BC34A; }\n        .simple-notification.alert { background: #ffdb5b; }\n        .simple-notification.info { background: #03A9F4; }\n        .simple-notification.warn { background: #ffdb5b; }\n\n        .simple-notification .sn-progress-loader {\n            position: absolute;\n            top: 0;\n            left: 0;\n            width: 100%;\n            height: 5px;\n        }\n\n        .simple-notification .sn-progress-loader span {\n            float: left;\n            height: 100%;\n        }\n\n        .simple-notification.success .sn-progress-loader span { background: #689F38; }\n        .simple-notification.error .sn-progress-loader span { background: #D32F2F; }\n        .simple-notification.alert .sn-progress-loader span { background: #edc242; }\n        .simple-notification.info .sn-progress-loader span { background: #0288D1; }\n        .simple-notification.warn .sn-progress-loader span { background: #edc242; }\n        .simple-notification.bare .sn-progress-loader span { background: #ccc; }\n\n        .simple-notification.warn div .sn-title { color: #444; }\n        .simple-notification.warn div .sn-content { color: #444; }\n    '];
export const RenderType_NotificationComponent:i0.RendererType2 = i0.ɵcrt({encapsulation:2,
    styles:styles_NotificationComponent,data:{'animation':[{type:7,name:'enterLeave',
        definitions:[{type:0,name:'fromRight',styles:{type:6,styles:{opacity:1,transform:'translateX(0)'},
            offset:(null as any)}},{type:1,expr:'* => fromRight',animation:[{type:6,
            styles:{opacity:0,transform:'translateX(5%)'},offset:(null as any)},{type:4,
            styles:(null as any),timings:'400ms ease-in-out'}],options:(null as any)},
            {type:0,name:'fromRightOut',styles:{type:6,styles:{opacity:0,transform:'translateX(-5%)'},
                offset:(null as any)}},{type:1,expr:'fromRight => fromRightOut',animation:[{type:6,
                styles:{opacity:1,transform:'translateX(0)'},offset:(null as any)},
                {type:4,styles:(null as any),timings:'300ms ease-in-out'}],options:(null as any)},
            {type:0,name:'fromLeft',styles:{type:6,styles:{opacity:1,transform:'translateX(0)'},
                offset:(null as any)}},{type:1,expr:'* => fromLeft',animation:[{type:6,
                styles:{opacity:0,transform:'translateX(-5%)'},offset:(null as any)},
                {type:4,styles:(null as any),timings:'400ms ease-in-out'}],options:(null as any)},
            {type:0,name:'fromLeftOut',styles:{type:6,styles:{opacity:0,transform:'translateX(5%)'},
                offset:(null as any)}},{type:1,expr:'fromLeft => fromLeftOut',animation:[{type:6,
                styles:{opacity:1,transform:'translateX(0)'},offset:(null as any)},
                {type:4,styles:(null as any),timings:'300ms ease-in-out'}],options:(null as any)},
            {type:0,name:'scale',styles:{type:6,styles:{opacity:1,transform:'scale(1)'},
                offset:(null as any)}},{type:1,expr:'* => scale',animation:[{type:6,
                styles:{opacity:0,transform:'scale(0)'},offset:(null as any)},{type:4,
                styles:(null as any),timings:'400ms ease-in-out'}],options:(null as any)},
            {type:0,name:'scaleOut',styles:{type:6,styles:{opacity:0,transform:'scale(0)'},
                offset:(null as any)}},{type:1,expr:'scale => scaleOut',animation:[{type:6,
                styles:{opacity:1,transform:'scale(1)'},offset:(null as any)},{type:4,
                styles:(null as any),timings:'400ms ease-in-out'}],options:(null as any)},
            {type:0,name:'rotate',styles:{type:6,styles:{opacity:1,transform:'rotate(0deg)'},
                offset:(null as any)}},{type:1,expr:'* => rotate',animation:[{type:6,
                styles:{opacity:0,transform:'rotate(5deg)'},offset:(null as any)},
                {type:4,styles:(null as any),timings:'400ms ease-in-out'}],options:(null as any)},
            {type:0,name:'rotateOut',styles:{type:6,styles:{opacity:0,transform:'rotate(-5deg)'},
                offset:(null as any)}},{type:1,expr:'rotate => rotateOut',animation:[{type:6,
                styles:{opacity:1,transform:'rotate(0deg)'},offset:(null as any)},
                {type:4,styles:(null as any),timings:'400ms ease-in-out'}],options:(null as any)}],
        options:{}}]}});
function View_NotificationComponent_2(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),0,'div',[['class',
      'icon']],[[8,'innerHTML',1]],(null as any),(null as any),(null as any),(null as any)))],
      (null as any),(_ck,_v) => {
        var _co:any = _v.component;
        const currVal_0:any = _co.safeSvg;
        _ck(_v,0,0,currVal_0);
      });
}
function View_NotificationComponent_1(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),11,'div',([] as any[]),
      (null as any),(null as any),(null as any),(null as any),(null as any))),(_l()(),
      i0.ɵted((null as any),['\n                '])),(_l()(),i0.ɵeld(0,(null as any),
      (null as any),1,'div',[['class','sn-title']],(null as any),(null as any),(null as any),
      (null as any),(null as any))),(_l()(),i0.ɵted((null as any),['',''])),(_l()(),
      i0.ɵted((null as any),['\n                '])),(_l()(),i0.ɵeld(0,(null as any),
      (null as any),2,'div',[['class','sn-content']],(null as any),(null as any),(null as any),
      (null as any),(null as any))),(_l()(),i0.ɵted((null as any),['',''])),i0.ɵppd(2),
      (_l()(),i0.ɵted((null as any),['\n\n                '])),(_l()(),i0.ɵand(16777216,
          (null as any),(null as any),1,(null as any),View_NotificationComponent_2)),
      i0.ɵdid(16384,(null as any),0,i1.NgIf,[i0.ViewContainerRef,i0.TemplateRef],{ngIf:[0,
          'ngIf']},(null as any)),(_l()(),i0.ɵted((null as any),['\n            ']))],
      (_ck,_v) => {
        var _co:any = _v.component;
        const currVal_2:any = (_co.item.icon !== 'bare');
        _ck(_v,10,0,currVal_2);
      },(_ck,_v) => {
        var _co:any = _v.component;
        const currVal_0:any = _co.item.title;
        _ck(_v,3,0,currVal_0);
        const currVal_1:any = i0.ɵunv(_v,6,0,_ck(_v,7,0,i0.ɵnov((<any>_v.parent),0),
            _co.item.content,_co.maxLength));
        _ck(_v,6,0,currVal_1);
      });
}
function View_NotificationComponent_3(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),0,'div',([] as any[]),
      [[8,'innerHTML',1]],(null as any),(null as any),(null as any),(null as any)))],
      (null as any),(_ck,_v) => {
        var _co:any = _v.component;
        const currVal_0:any = _co.item.html;
        _ck(_v,0,0,currVal_0);
      });
}
function View_NotificationComponent_4(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),5,'div',[['class',
      'sn-progress-loader']],(null as any),(null as any),(null as any),(null as any),
      (null as any))),(_l()(),i0.ɵted((null as any),['\n                '])),(_l()(),
      i0.ɵeld(0,(null as any),(null as any),2,'span',([] as any[]),(null as any),(null as any),
          (null as any),(null as any),(null as any))),i0.ɵdid(278528,(null as any),
      0,i1.NgStyle,[i0.KeyValueDiffers,i0.ElementRef,i0.Renderer],{ngStyle:[0,'ngStyle']},
      (null as any)),i0.ɵpod(['width']),(_l()(),i0.ɵted((null as any),['\n            ']))],
      (_ck,_v) => {
        var _co:any = _v.component;
        const currVal_0:any = _ck(_v,4,0,(_co.progressWidth + '%'));
        _ck(_v,3,0,currVal_0);
      },(null as any));
}
export function View_NotificationComponent_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[i0.ɵpid(0,i2.MaxPipe,([] as any[])),(_l()(),i0.ɵted((null as any),
      ['\n        '])),(_l()(),i0.ɵeld(0,(null as any),(null as any),12,'div',[['class',
      'simple-notification']],[[24,'@enterLeave',0]],[[(null as any),'click'],[(null as any),
      'mouseenter'],[(null as any),'mouseleave']],(_v,en,$event) => {
    var ad:boolean = true;
    var _co:i3.NotificationComponent = _v.component;
    if (('click' === en)) {
      const pd_0:any = ((<any>_co.onClick($event)) !== false);
      ad = (pd_0 && ad);
    }
    if (('mouseenter' === en)) {
      const pd_1:any = ((<any>_co.onEnter()) !== false);
      ad = (pd_1 && ad);
    }
    if (('mouseleave' === en)) {
      const pd_2:any = ((<any>_co.onLeave()) !== false);
      ad = (pd_2 && ad);
    }
    return ad;
  },(null as any),(null as any))),i0.ɵdid(278528,(null as any),0,i1.NgClass,[i0.IterableDiffers,
      i0.KeyValueDiffers,i0.ElementRef,i0.Renderer],{klass:[0,'klass'],ngClass:[1,
      'ngClass']},(null as any)),i0.ɵpod(['alert','error','warn','success','info',
      'bare','rtl-mode']),(_l()(),i0.ɵted((null as any),['\n\n            '])),(_l()(),
      i0.ɵand(16777216,(null as any),(null as any),1,(null as any),View_NotificationComponent_1)),
      i0.ɵdid(16384,(null as any),0,i1.NgIf,[i0.ViewContainerRef,i0.TemplateRef],{ngIf:[0,
          'ngIf']},(null as any)),(_l()(),i0.ɵted((null as any),['\n            '])),
      (_l()(),i0.ɵand(16777216,(null as any),(null as any),1,(null as any),View_NotificationComponent_3)),
      i0.ɵdid(16384,(null as any),0,i1.NgIf,[i0.ViewContainerRef,i0.TemplateRef],{ngIf:[0,
          'ngIf']},(null as any)),(_l()(),i0.ɵted((null as any),['\n\n            '])),
      (_l()(),i0.ɵand(16777216,(null as any),(null as any),1,(null as any),View_NotificationComponent_4)),
      i0.ɵdid(16384,(null as any),0,i1.NgIf,[i0.ViewContainerRef,i0.TemplateRef],{ngIf:[0,
          'ngIf']},(null as any)),(_l()(),i0.ɵted((null as any),['\n\n        '])),
      (_l()(),i0.ɵted((null as any),['\n    ']))],(_ck,_v) => {
    var _co:i3.NotificationComponent = _v.component;
    const currVal_1:any = _co.theClass;
    const currVal_2:any = _ck(_v,4,0,(_co.item.type === 'alert'),(_co.item.type === 'error'),
        (_co.item.type === 'warn'),(_co.item.type === 'success'),(_co.item.type === 'info'),
        (_co.item.type === 'bare'),_co.rtl);
    _ck(_v,3,0,currVal_1,currVal_2);
    const currVal_3:boolean = !_co.item.html;
    _ck(_v,7,0,currVal_3);
    const currVal_4:any = _co.item.html;
    _ck(_v,10,0,currVal_4);
    const currVal_5:any = _co.showProgressBar;
    _ck(_v,13,0,currVal_5);
  },(_ck,_v) => {
    var _co:i3.NotificationComponent = _v.component;
    const currVal_0:any = _co.item.state;
    _ck(_v,2,0,currVal_0);
  });
}
export function View_NotificationComponent_Host_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),1,'simple-notification',
      ([] as any[]),(null as any),(null as any),(null as any),View_NotificationComponent_0,
      RenderType_NotificationComponent)),i0.ɵdid(245760,(null as any),0,i3.NotificationComponent,
      [i4.NotificationsService,i5.DomSanitizer,i0.NgZone],(null as any),(null as any))],
      (_ck,_v) => {
        _ck(_v,1,0);
      },(null as any));
}
export const NotificationComponentNgFactory:i0.ComponentFactory<i3.NotificationComponent> = i0.ɵccf('simple-notification',
    i3.NotificationComponent,View_NotificationComponent_Host_0,{timeOut:'timeOut',
        showProgressBar:'showProgressBar',pauseOnHover:'pauseOnHover',clickToClose:'clickToClose',
        maxLength:'maxLength',theClass:'theClass',rtl:'rtl',animate:'animate',position:'position',
        item:'item'},{},([] as any[]));
